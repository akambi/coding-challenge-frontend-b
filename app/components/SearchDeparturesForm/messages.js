/*
 * This contains all the text for the SearchDeparturesForm component.
 */
import { defineMessages } from 'react-intl';

export const scope = 'busbud.components.SearchDeparturesForm';

export default defineMessages({
  origin: {
    id: `${scope}.origin`,
    defaultMessage: 'Origin',
  },
  destination: {
    id: `${scope}.destination`,
    defaultMessage: 'Destination',
  },
  outboundDate: {
    id: `${scope}.outboundDate`,
    defaultMessage: 'Date',
  },
  adult: {
    id: `${scope}.adult`,
    defaultMessage: 'Adults',
  },
  quebec: {
    id: `${scope}.quebec`,
    defaultMessage: 'Québec',
  },
  montreal: {
    id: `${scope}.montreal`,
    defaultMessage: 'Montréal',
  },
  chooseUpcomingDate: {
    id: `${scope}.chooseUpcomingDate`,
    defaultMessage: 'Date must be in the future',
  },
  search: {
    id: `${scope}.search`,
    defaultMessage: 'Search',
  },
});
