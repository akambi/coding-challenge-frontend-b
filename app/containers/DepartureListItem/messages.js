/*
 * This contains all the text for the DepartureListItem component.
 */
import { defineMessages } from 'react-intl';

export const scope = 'busbud.components.DepartureListItem';

export default defineMessages({
  quebec: {
    id: `${scope}.quebec`,
    defaultMessage: 'Québec',
  },
  montreal: {
    id: `${scope}.montreal`,
    defaultMessage: 'Montréal',
  },
});
