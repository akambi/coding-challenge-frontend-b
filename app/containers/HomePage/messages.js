/*
 * HomePage Messages
 *
 * This contains all the text for the HomePage component.
 */
import { defineMessages } from 'react-intl';

export const scope = 'busbud.containers.HomePage';

export default defineMessages({
  searchDeparturesHeader: {
    id: `${scope}.searchDeparturesHeader`,
    defaultMessage:
      'Search all departures for a given origin city and a given destination city for a given day',
  },
});
