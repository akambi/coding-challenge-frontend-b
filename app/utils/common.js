export const formatTime = date => new Date(date).toLocaleTimeString();
