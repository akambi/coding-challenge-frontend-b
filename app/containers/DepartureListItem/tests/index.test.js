/**
 * Test the departure list item
 */

import React from 'react';
import { getByText, render } from 'react-testing-library';
import { IntlProvider } from 'react-intl';
import { formatTime } from 'utils/common';

import { DepartureListItem } from '../index';
import messages from '../messages';

const renderComponent = (props = {}) =>
  render(
    <IntlProvider locale="en">
      <DepartureListItem {...props} />
    </IntlProvider>,
  );

describe('<DepartureListItem />', () => {
  let searchParams;
  let item;

  // Before each test reset the item data for safety
  beforeEach(() => {
    searchParams = {
      adult: 1,
      origin: 'quebec',
      destination: 'montreal',
      outboundDate: '2020-12-14T14:30:12.149Z',
    };

    item = {
      arrival_time: '2020-12-14T14:15:00',
      departure_time: '2020-12-14T10:30:00',
      location: {
        id: 12189,
        city_id: '375dd587-9001-acbd-84a4-683deddcb1b1',
        name: 'Gare du Palais',
        address: ['320 Rue Abraham Martin', 'QuÃ©bec, QC G1K 3X2', 'Canada'],
        type: 'bus_station',
      },
      price: 5524,
    };
  });

  it('should render a ListItem', () => {
    const { container } = renderComponent({ searchParams, item });
    expect(container.firstChild).toMatchSnapshot();
  });

  it('should render the departure city', () => {
    const { queryByText } = renderComponent({
      searchParams,
      item,
    });
    const origin = messages[searchParams.origin].defaultMessage;
    expect(queryByText(origin)).not.toBeNull();
  });

  it('should render the arrival city', () => {
    const { queryByText } = renderComponent({
      searchParams,
      item,
    });

    const destination = messages[searchParams.destination].defaultMessage;
    expect(queryByText(destination)).not.toBeNull();
  });

  it('should render the location name', () => {
    const { container } = renderComponent({ searchParams, item });
    expect(
      getByText(container, content => content.endsWith(item.location.name)),
    ).not.toBeNull();
  });

  it('should render the departure time', () => {
    const { container } = renderComponent({ searchParams, item });
    expect(
      getByText(container, formatTime(item.departure_time)),
    ).not.toBeNull();
  });

  it('should render the arrival time', () => {
    const { container } = renderComponent({ searchParams, item });
    expect(getByText(container, formatTime(item.arrival_time))).not.toBeNull();
  });
});
