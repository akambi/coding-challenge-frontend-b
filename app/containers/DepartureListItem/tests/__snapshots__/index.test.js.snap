// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<DepartureListItem /> should render a ListItem 1`] = `
<li
  class="Wrapper-sc-11n0iqq-0 fJThNy"
>
  <div
    class="Item-sc-141m8an-0 gvGHMp"
  >
    <div
      class="Wrapper-sc-1k2q7aa-0 dtddjJ"
    >
      <div>
        <strong>
          10:30:00
           
        </strong>
        <span>
          Québec
        </span>
        <span>
           - 
          Gare du Palais
        </span>
      </div>
      <div>
        <strong>
          14:15:00
           
        </strong>
        <span>
          Montréal
        </span>
      </div>
      <div>
        <strong>
          <span>
            CA$55.24
          </span>
        </strong>
      </div>
    </div>
  </div>
</li>
`;
